app/screens/edit_channel/edit_channel.js

```diff
@@ -114,32 +115,42 @@ export default class EditChannel extends PureComponent {
         this.emitCanUpdateChannel(false);
     }
 
-    componentWillReceiveProps(nextProps) {
-        if (this.props.theme !== nextProps.theme) {
-            setNavigatorStyles(this.props.componentId, nextProps.theme);
-        }
-
+    static getDerivedStateFromProps(nextProps, state) {
         const {updateChannelRequest} = nextProps;
 
-        if (this.props.updateChannelRequest !== updateChannelRequest) {
+        if (state.updateChannelRequest !== updateChannelRequest) {
+            const newState = {
+                error: null,
+                updating: true,
+            };
+
             switch (updateChannelRequest.status) {
             case RequestStatus.STARTED:
                 this.emitUpdating(true);
```

Comment:
`this` is no longer defined, so this will need to be moved to the `componentDidUpdate`. I think the same thing needs to happen for the calls to `EventEmitter` and such below as well

Reply:
Hi @hmhealey thanks for your suggestions. I have a question: do we need `getDerivedStateFromProps` here because there can be a case when a user is trying to edit a channel and someone has already updated the channel?

Reply:
No, I don't think we need to handle that case here. `updateChannelRequest` only tracks requests made from this device, so we don't see anything from other users even if we use `getDerivedStateFromProps`